qutebrowser/completion/models/urlmodel.py

```diff
@@ -126,3 +155,50 @@ def on_quickmark_removed(self, name):
             if name_item.data(Qt.DisplayRole) == name:
                 self._quickmark_cat.removeRow(i)
                 break
+
+    @pyqtSlot(str, str)
+    def on_bookmark_added(self, title, url):
+        """Called when a bookmark has been added by the user.
+
+        Args:
+            title: The title of the new bookmark.
+            url: The url of the new bookmark, as string.
+        """
+        self._add_bookmark_entry(title, url)
+
+    @pyqtSlot(str)
+    def on_bookmark_removed(self, url):
+        """Called when a bookmark has been removed by the user.
+
+        Args:
+            url: The url of the bookmark which has been removed.
+        """
+        for i in range(self._bookmark_cat.rowCount()):
+            url_item = self._bookmark_cat.child(i, 0)
+            if url_item.data(Qt.DisplayRole) == url:
+                self._bookmark_cat.removeRow(i)
+                break
+
+    def delete_cur_item(self, win_id):
+        """Delete the selected item.
+
+        Args:
+            win_id: The current windows id.
+        """
+        completion = objreg.get('completion', scope='window',
+                                window=win_id)
+        index = completion.currentIndex()
+        model = completion.model()
+        url = model.data(index)
+        category = index.parent()
+        if category.isValid():
+            if category.data() == 'Bookmarks':
+                bookmark_manager = objreg.get('bookmark-manager')
+                bookmark_manager.delete(url)
+                message.info(win_id, "Bookmarks deleted")
+            elif category.data() == 'Quickmarks':
+                quickmark_manager = objreg.get('quickmark-manager')
+                name = model.data(index.sibling(index.row(),
+                                  index.column() + 1))
```

Comment:
Why `index.column() + 1` and not just `1` here? I'm not sure if it's always the leftmost element will be selected, but the titles will always be in column 1, right?